x/lockup/keeper/lock.go

```diff
@@ -300,10 +300,8 @@ func (k Keeper) addTokensToLock(ctx sdk.Context, lock *types.PeriodLock, coins s
 	}
 
 	// modifications to accumulation store
-	for _, coin := range lock.Coins {
-		// remove previous store for the lock ID and add again with updated value
-		k.accumulationStore(ctx, coin.Denom).Remove(accumulationKey(lock.Duration, lock.ID))
-		k.accumulationStore(ctx, coin.Denom).Set(accumulationKey(lock.Duration, lock.ID), coin.Amount)
+	for _, coin := range coins {
+		k.accumulationStore(ctx, coin.Denom).Increase(accumulationKey(lock.Duration), coin.Amount)
```

Comment:
Oh wow, looks like a bug before lol

Reply:
I think it was fine on current code. Just something that didn't exist when the PR was made. It was just replacing the whole lock accumulation obj and adding each coin in the new lock. Whereas now it's only increasing the accumulator by what is being added to the existing lock.

It would be a bug I think tho if I had left for _, coin := range lock.Coins and just changed the accumulationStore call, since then it would be adding the whole lock (duplicating the existing part)

Reply:
ah your right